Restore WC_Logger remove and clear functionality

sirreal opened the pull request:
`WC_Logger::clear` and `WC_Logger::remove` were deprecated in favor of the `WC_Log_Handler_File` methods of the same name. The functionality was also removed.

This restores the functionality to improve compatibility with existing code.

sirreal committed:
Restore remove and clear functionality to WC_Logger

Maintain backwards compatibility of removing or clearing files.

sirreal committed:
Test deprecated WC_Logger::clear functionality

sirreal committed:
Add test for deprecated WC_Logger::remove

claudiosanches commented on includes/class-wc-logger.php:
```diff
@@ -216,22 +216,28 @@ public function debug( $message, $context = array() ) {
+	 *
 	 * @return bool
 	 */
-	public function remove() {
```
Sorry, but I just saw that `remove()` is new in 2.7, so we can just remove it.

sirreal replied:
:man_facepalming: You're right, it was added post `2.6` in 26353d7099d270064bee97ef11acdcd4f6390d13.

👍 Removed.

sirreal committed:
Remove WC_Logger::remove

This method was added post 2.6 and does not need to be maintained/deprecated.
Remove related test.
